subjects/corewar/README.md

```diff
@@ -0,0 +1,366 @@
+
+To avoid infinite games, `CYCLES_TO_DIE` will be decremented by `CYCLE_DELTA` under certain conditions:
+
+- If during the last life loop there were at least `NBR_LIVE` successfully executed by the players.
+- If it has been `MAX_CHECKS` life loops since it was decremented last time.
+
+> Notice that some smart players may trick another player to start making `live` statements for them, this virtually means a player can still make `live` statements after all its processes were killed.
+
+### The Assembly language
+
+In the assembly file there must be a `.name "[Name of the player]"` field and a `.description "[Description of the player]"` before any instruction.
+
+Every instruction has to be on a single line and may be preceded by a label declaration.
+The instruction name is separated from the optional label and the parameters by one or more spaces.
+
+The parameters are separated from each other by `,`.
+
+The instructions will be of different sizes depending on the number of type of parameters passed.
+
+#### Labels
+
+Labels are also called pseudo instructions, in the sense they won't be translated into binary code.
+They are widely used to help programmers creating assembly code without needing to remember and count relative memory positions.
+
+So when looking in the binaries the labels will be replaced in order to match their relative position in bytes from the place they were used.
+
+As an example in the `ameba.s` player `zjmp %:hello` is exactly the same of `zjmp %-5` (`-5` being `ff fb` in hexadecimal notation).
+
+This is because from the `zjmp` you will need to go back 5 bytes in order to match the label `hello:` declaration.
+
+> Notice that labels can be declared at the start of an instruction or on their own line.
+
+#### Parameters types
+
+- Register parameter
+
+Each process has its own registers (from 1 to `REG_NUMBER`).
+Each of those registers can hold 32 bits, but the register number itself is written in only 8 bits (which are enough to go from 1 to 16).
+
+They are formatted as `r1`, `r2`, `...`, `r16`.
+
+- Direct parameter
+
+A direct parameter is a value that represents itself, it is formatted by using `%` before the value. So `%10` will represent the number ten.
+
+- Indirect parameter
+
+An indirect parameter will give you the relative position of the address where to look for the value, think of it as an ancestor of raw pointers.
+
+In this case `10` will say to the VM to look ten bytes forward the current instruction and take the value from there.
+
+> Don't be scared to use the provided Asm and VM to play with those different parameters in order to be sure you understood properly their behavior.
+
+#### Instruction Set
+
+| Name  | Nb Params | Opcode | Nb Cycles | Has Pcode | Has Idx | Params                                                                     |
```

Comment:
It might be a bit repetitive, but I would add a short description for each column: some information is already present in the below paragraph while some are missing, for example, `Nb Params` or `Params`. I am convinced is it possible to understand what each of the column means, but it would help reducing doubts